nomad/structs/structs.go

```diff
@@ -6581,12 +6570,10 @@ func validateServices(t *Task) error {
 
 	// Get the set of port labels.
 	portLabels := make(map[string]struct{})
-	if t.Resources != nil {
-		for _, network := range t.Resources.Networks {
-			ports := network.PortLabels()
-			for portLabel := range ports {
-				portLabels[portLabel] = struct{}{}
-			}
+	if len(tgNetworks) > 0 {
+		ports := tgNetworks[0].PortLabels()
+		for portLabel := range ports {
+			portLabels[portLabel] = struct{}{}
 		}
 	}
```

Comment:
Is there a backward compatibility concern here?  If an old job that specifies ports and labels in task resources, will they continue to be validated/processed the same way?

Reply:
Checking here again - if the answer is no, let's merge then.

Reply:
@notnoop I've re-added the removed block, appending task level port labels to the set of port labels to validate against.